tempesta_fw/classifier/req_conn_limit.c

```diff
@@ -251,11 +257,109 @@ rcl_req_limit(RclAccount *ra, struct sock *sk)
 }
 
 static int
+rcl_http_uri_len_limit(const TfwHttpReq *req)
+{
+	/* FIXME: tfw_str_len() iterates over chunks to calculate the length.
+	 * This is too slow. The value must be stored in a TfwStr field. */
+	if (rcl_http_uri_len && tfw_str_len(&req->uri) > rcl_http_uri_len) {
+		TFW_DBG("rcl: http_uri_len limit is reached\n");
+		return TFW_BLOCK;
+	}
+
+	return TFW_PASS;
+}
+
+static int
+rcl_http_field_len_limit(const TfwHttpReq *req)
+{
+	const TfwStr *field, *end;
+
+	if (!rcl_http_field_len)
+		return TFW_PASS;
+
+	TFW_HTTP_FOR_EACH_HDR_FIELD(field, end, req) {
+		if (tfw_str_len(field) > rcl_http_field_len) {
+			TFW_DBG("rcl: http_field_len limit is reached\n");
+			return TFW_BLOCK;
+		}
+	}
```

Comment:
So Cookie is also limited by this routine while Cookie is very specific header. Usually, we can limit HTTP header by 256 characters or something like that, while common Cookie is much larger. Cookies must be limited separately.